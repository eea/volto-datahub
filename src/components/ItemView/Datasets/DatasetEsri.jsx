import { List } from 'semantic-ui-react';
import { SVGIcon } from '../utils.js';

import lockSVG from 'remixicon/icons/System/lock-line.svg';
import { isInternalURL } from '@eeacms/volto-datahub/utils';
import servicesSVG from '../icons/services.svg';

const ItemLink = ({ item }) => (
  <List.Item key={item.id}>
    <List.Content>
      <div className="dataset-item">
        <SVGIcon name={servicesSVG} size="20px" />
        <a className="item-link" href={item.url} target="_blank" rel="noopener">
          {item.protocol} {item.name || item.description}
        </a>
        {isInternalURL(item.url) ? (
          <SVGIcon name={lockSVG} size="20px" />
        ) : null}
      </div>
    </List.Content>
  </List.Item>
);

export default function DatasetEsri({ dataset }) {
  return dataset['ESRI:REST'] || dataset['OGC:WMS'] ? (
    <>
      <h5>Services:</h5>
      <List divided relaxed>
        {[...(dataset['ESRI:REST'] || []), ...(dataset['OGC:WMS'] || [])].map(
          (item, i) => (
            <ItemLink item={item} key={item.id} />
          ),
        )}
      </List>
    </>
  ) : null;
}
