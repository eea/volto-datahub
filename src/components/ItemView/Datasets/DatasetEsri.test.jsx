import React from 'react';
import renderer from 'react-test-renderer';

import DatasetEsri from './DatasetEsri';

jest.mock('@eeacms/search', () => ({
  runRequest: jest.fn(),
}));

const dataset = {
  'ESRI:REST': [
    {
      id: 1,
      url: 'https://example.org/1',
      protocol: 'https',
      name: 'First',
      description: 'Something here',
    },
  ],
};

describe('DatasetEsri', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<DatasetEsri dataset={dataset} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
