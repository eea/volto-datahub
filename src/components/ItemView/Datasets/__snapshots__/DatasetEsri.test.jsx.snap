// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DatasetEsri renders correctly 1`] = `
Array [
  <h5>
    Services:
  </h5>,
  <div
    className="ui divided relaxed list"
    role="list"
  >
    <div
      className="item"
      onClick={[Function]}
      role="listitem"
    >
      <div
        className="content"
      >
        <div
          className="dataset-item"
        >
          <svg
            className="icon"
            dangerouslySetInnerHTML={
              Object {
                "__html": undefined,
              }
            }
            style={
              Object {
                "fill": "currentColor",
                "height": "20px",
                "width": "auto",
              }
            }
          />
          <a
            className="item-link"
            href="https://example.org/1"
            target="_blank"
          >
            https
             
            First
          </a>
        </div>
      </div>
    </div>
  </div>,
]
`;
